### PR TITLE
Made changes needed for publisher CRUD features to work on frontend

### PR DESCRIPTION
**Pull request by justin-obrien-smoothstack**
Despite the branch name, this includes everything needed for read and delete as well.

### CODE REVIEW

File: service/publisherCrudService.js [hunk 1]
```diff
@@ -28,25 +34,91 @@ exports.createPublisher = (publisher) => {
...
+            publisher.publisherId,
+            false
+          );
+        } catch (error) {
+          results.readBooksError = true;
+          db.rollback(() => reject(results));
+          return;
+        }
+        publisher.bookIds = [];
+        for (const book of books) publisher.bookIds.push(book.bookId);
```
Comment (trevorhuis-smoothstack) [inline service/publisherCrudService.js:117]: Why did you decide to only get the book IDs and not the titles while you were querying that table?

**Review by trevorhuis-smoothstack — APPROVED** [review]
I see the name of the PR is to get these functions to work on the front end. Did you make them all async/await because the methods weren't working with then/catch or to make all the functions consistent?
Or were the try/catch blocks easier to work with now that you are also using the bookDAO?

File: service/publisherCrudService.js (hunk 1 above)
Comment (justin-obrien-smoothstack) [reply service/publisherCrudService.js:117]: It feels more elegant to me. An object carries the primary keys of its related objects, which is the minimum information needed to identify those objects unambiguously. Any other information about the objects can be accessed using the primary keys. It avoids issues such as, what if someone tries to create/update a publisher and they include books where the title doesn't match the primary key.

It also uses less memory to store a publisher object, but of course the tradeoff there is the extra resources that must be used to get book information other than primary keys.

**Comment by justin-obrien-smoothstack** [comment]
I prefer async/await over then/catch. I think it makes for cleaner code, although it can take a little more getting used to. I just used then/catch in the first place because we were supposed to use all the different techniques.

**Pull request merged**